crates/sui-core/src/authority_server.rs

```diff
@@ -1310,49 +1311,27 @@ impl ValidatorService {
                             });
                         }
                     }
-                },
-                mut effects = self.state
-                    .get_transaction_cache_reader()
-                    .notify_read_executed_effects("AuthorityServer::notify_read_executed_effects", &tx_digests) => {
+                }
+
+                mut outputs = self.state.get_transaction_cache_reader().notify_read_fastpath_transaction_outputs(tx_digests),
+                    if current_status == Some(ConsensusTxStatus::FastpathCertified) => {
```

Comment:
Assuming I get this correctly, the `wait_for_fastpath_effects` it does literally care only for returning effects that are coming from the fast path. If the transaction gets `Finalised` (either via the FP commit output or via QD) this method will not return anything (until an Expired) status is finally casted by the cache. I guess via FP we will always cast a `FastpathCertified` status, before a `Finalised` one, so it should be really the QD scenario.